src/error.rs

```diff
@@ -6,55 +6,56 @@
 /// request guards' fields).
 ///
 /// HandlerErrors are rocket Responders (render their own error responses).
-use std::fmt;
 use std::result;
 
-use failure::{Backtrace, Context, Error, Fail};
+//use failure::{Backtrace, Context, Error, Fail};
 use rocket::http::Status;
 use rocket::response::{Responder, Response};
 use rocket::{response, Request};
 use rocket_contrib::{json, json::Json};
+use thiserror::Error;
 
-pub type Result<T> = result::Result<T, Error>;
+pub type Result<T> = result::Result<T, HandlerError>;
 
 pub type HandlerResult<T> = result::Result<T, HandlerError>;
 
 /// Signal a request guard failure, propagated up to the Handler to render an
 /// error response
 pub const VALIDATION_FAILED: Status = Status::InternalServerError;
-
-#[derive(Debug)]
+/*
+#[derive(Debug, Error)]
 pub struct HandlerError {
     inner: Context<HandlerErrorKind>,
 }
+*/
 
-#[derive(Clone, Eq, PartialEq, Debug, Fail)]
-pub enum HandlerErrorKind {
+#[derive(Clone, Eq, PartialEq, Debug, Error)]
+pub enum HandlerError {
```

Comment:
Note that this new error will not have a backtrace. It would need to do something like [this](https://github.com/mozilla-services/autopush-rs/blob/761d91f4fc06176dbc1ddc0d654d2b1981eebe81/autoendpoint/src/error.rs#L31) (keeping the wrapper struct) if backtraces are still desired.

Reply:
Ah, thanks. I wasn't quite sure how that worked with the new config.

Reply:
It's a little hidden, but here's where the backtrace is created in autoendpoint-rs:
https://github.com/mozilla-services/autopush-rs/blob/761d91f4fc06176dbc1ddc0d654d2b1981eebe81/autoendpoint/src/error.rs#L218-L230